modules/copybara/0.0.0-20240326-6485af6/patches/update_repos.patch

```diff
@@ -0,0 +1,228 @@
+ load("//third_party:bazel_skylib.bzl", "skylib_sha256", "skylib_version")
+
+ def copybara_repositories():
+@@ -39,27 +38,6 @@ def copybara_repositories():
+         # sha256 = RULES_JVM_EXTERNAL_SHA,
+         strip_prefix = "rules_jvm_external-%s" % RULES_JVM_EXTERNAL_TAG,
+         url = "https://github.com/bazelbuild/rules_jvm_external/archive/%s.zip" % RULES_JVM_EXTERNAL_TAG,
+-        patches = ["@io_bazel//third_party:rules_jvm_external_6.0.patch"],
+-        patch_args = ["-p1"],
+-    )
+-
+-    # LICENSE: The Apache Software License, Version 2.0
+-    maybe(
+-        http_archive,
+-        name = "io_bazel",
+-        sha256 = bazel_sha256,
+-        strip_prefix = "bazel-" + bazel_version,
+-        url = "https://github.com/bazelbuild/bazel/archive/" + bazel_version + ".zip",
+-    )
+-
+-    # LICENSE: The Apache Software License, Version 2.0
+-    # Buildifier and friends:
+-    maybe(
+-        http_archive,
+-        name = "buildtools",
+-        sha256 = buildtools_sha256,
+-        strip_prefix = "buildtools-" + buildtools_version,
+-        url = "https://github.com/bazelbuild/buildtools/archive/" + buildtools_version + ".zip",
+     )
+
+     # LICENSE: The Apache Software License, Version 2.0
+@@ -173,6 +151,33 @@ def copybara_repositories():
+         ],
+     )
+
++    # LICENSE: MIT
++    maybe(
++        http_archive,
++        name = "buildifier_prebuilt",
++        sha256 = "8ada9d88e51ebf5a1fdff37d75ed41d51f5e677cdbeafb0a22dda54747d6e07e",
++        strip_prefix = "buildifier-prebuilt-6.4.0",
++        urls = [
++            "http://github.com/keith/buildifier-prebuilt/archive/6.4.0.tar.gz",
++        ],
++    )
++
++    _non_module_deps(None)
++
++def _non_module_deps(_):
++    # LICENSE: The Apache Software License, Version 2.0
++    maybe(
++        http_archive,
++        name = "io_bazel",
++        sha256 = bazel_sha256,
++        strip_prefix = "bazel-" + bazel_version,
++        url = "https://github.com/bazelbuild/bazel/archive/" + bazel_version + ".zip",
++        patch_args = ["-p1"],
++        patches = [
++            "//third_party/bazel:bazel.patch",
++        ],
++    )
++
+     # LICENSE: The Apache Software License, Version 2.0
+     maybe(
+         http_archive,
+@@ -183,3 +188,5 @@ def copybara_repositories():
+         ],
+         build_file = Label("//external/third_party:jcommander.BUILD"),
+     )
++
++non_module_deps = module_extension(implementation = _non_module_deps)
+diff --git a/third_party/bazel/bazel.patch b/third_party/bazel/bazel.patch
+new file mode 100644
+index 00000000..30a50b8b
+--- /dev/null
++++ b/third_party/bazel/bazel.patch
+@@ -0,0 +1,142 @@
++diff --git a/third_party/BUILD b/third_party/BUILD
++index 262577182a..1bf4666b53 100644
++--- a/third_party/BUILD
+++++ b/third_party/BUILD
++@@ -252,4 +252,4 @@ distrib_jar_filegroup(
++     srcs = [
++-        "@maven//:com_google_auto_value_auto_value_annotations_file",
```

Comment:
Do we really need this patch file? I thought those `@maven//:*_file` targets won't be used by the targets copybara depends on.

Reply:
i think we can get away with just this one:

```bzl
diff --git a/third_party/BUILD b/third_party/BUILD
index 262577182a..6300c8f25a 100644
--- a/third_party/BUILD
+++ b/third_party/BUILD
@@ -352,17 +352,15 @@ filegroup(
 
 # TODO(https://github.com/bazelbuild/bazel/issues/18214): After fixing Guava leak
 # in test-runner, the guava target can be reverted back to java_library
-java_import(
+java_library(
     name = "guava",
     applicable_licenses = [":guava_license"],
-    jars = [
-        "@maven//:com_google_guava_guava_file",
-        "@maven//:com_google_guava_failureaccess_file",
-    ],
     exports = [
         ":error_prone_annotations",
         ":jcip_annotations",
         ":jsr305",
+        "@maven//:com_google_guava_failureaccess",
+        "@maven//:com_google_guava_guava",
     ],
 )

```

Reply:
commented on that upstream issue to see if we can remove that from bazel